patches/minecraft/net/minecraft/item/SpawnEggItem.java.patch

```diff
@@ -0,0 +1,55 @@
+--- a/net/minecraft/item/SpawnEggItem.java
++++ b/net/minecraft/item/SpawnEggItem.java
+@@ -35,18 +35,28 @@
+ 
+ public class SpawnEggItem extends Item {
+    public static final Map<EntityType<?>, SpawnEggItem> field_195987_b = Maps.newIdentityHashMap();
++   public static final java.util.List<SpawnEggItem> MOD_EGGS = new java.util.ArrayList<>();
+    private final int field_195988_c;
+    private final int field_195989_d;
+-   private final EntityType<?> field_200890_d;
++   private final java.util.function.Supplier<? extends EntityType<?>> field_200890_d;
+ 
++   /**@deprecated Use {@link SpawnEggItem#SpawnEggItem(java.util.function.Supplier, int, int, Item.Properties)}*/@Deprecated
+    public SpawnEggItem(EntityType<?> p_i48465_1_, int p_i48465_2_, int p_i48465_3_, Item.Properties p_i48465_4_) {
+       super(p_i48465_4_);
+-      this.field_200890_d = p_i48465_1_;
++      this.field_200890_d = () -> p_i48465_1_;
```

Comment:
This should be using the delegate, though because a bunch of mods currently pass null as the entity type it probably will need to end up as:
```java
this.typeIn = typeIn == null ? () -> null : typeIn.delegate;//TODO - 1.17: Remove check for null and enforce it being nonnull
```